### PR TITLE
Added query parameters support for java template

### PR DESCRIPTION
**Pull request by richierich1610**
This changes add functionality to parse request query paramaters for java template.
Signed-off-by: Richhpal Gora <richhpalgora1610@gmail.com>

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Which issue(s) this PR fixes 

Fixes #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: template/java8/entrypoint/src/main/java/com/openfaas/entrypoint/App.java [hunk 1]
```diff
@@ -73,8 +74,9 @@ public void handle(HttpExchange t) throws IOException {
             //     System.out.println("Req header " + entry.getKey() + " " + entry.getValue());
             // }
 
-            IRequest req = new Request(requestBody, reqHeadersMap);
-
+            String queryRaw = t.getRequestURI().getRawQuery();
+            IRequest req = new Request(requestBody, reqHeadersMap,getQueryParameters(queryRaw),queryRaw);
```
Comment (alexellis) [inline template/java8/entrypoint/src/main/java/com/openfaas/entrypoint/App.java]: What if we moved the method "getQueryParameters" into the Request class, would that make sense?

Comment (richierich1610) [reply template/java8/entrypoint/src/main/java/com/openfaas/entrypoint/App.java]: Yeah, we can do that. Here I am considering Request as a model and all injection is done by server. In this case , App is responsible to inject params in request model.
thoughts?

File: template/java8/model/src/main/java/com/openfaas/model/IRequest.java [hunk 2]
```diff
@@ -9,5 +9,7 @@
     String getBody();
     Map<String, String> getHeaders();
     String getHeader(String key);
+    String getQueryRaw();
+    Map<String, String> getQueryParameters();
```
Comment (alexellis) [inline template/java8/model/src/main/java/com/openfaas/model/IRequest.java]: I'm thinking this should just be `getQuery()`?

File: template/java8/model/src/main/java/com/openfaas/model/Request.java [hunk 3]
```diff
@@ -30,4 +42,42 @@ public String getHeader(String key) {
...
+		return queryRaw;
+	}
+
+    @Override
+	public Map<String, String> getQueryParameters() {
+    	return queryParameters;
+    }
+    
+	private Map<String, String> parseQueryParameters() {
+		Map<String, String> reqParametersMap = new HashMap<String, String>();
```
Comment (alexellis) [inline template/java8/model/src/main/java/com/openfaas/model/Request.java]: Is there no utility method within Java to do this operation?

File: template/java8/model/src/test/java/RequestTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import static org.junit.Assert.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.openfaas.model.Request;
+public class RequestTest {
+
+	@Test
+	public void testRequestParametersGetSet()
```
Comment (alexellis) [inline template/java8/model/src/test/java/RequestTest.java]: I'd like to see:

- a test which has multiple parameters
- a test which has no parameters at all
- a test which has an empty parameter

File: template/java8/model/src/main/java/com/openfaas/model/Request.java [hunk 5]
```diff
@@ -4,16 +4,28 @@
...
     }
+    
+    public Request(String body, Map<String, String> headers,String queryRaw) {
+        this.body = body;
+        this.headers = headers;
+        this.queryRaw = queryRaw;
+        this.queryParameters = this.parseQueryParameters();
+    }
 
     public String getBody() {
```
Comment (alexellis) [inline template/java8/model/src/main/java/com/openfaas/model/Request.java]: Does it need `@Override`?

Comment (richierich1610) [reply template/java8/model/src/main/java/com/openfaas/model/Request.java]: @Override is not needed! Its good to have on methods to indicate that method is overridden from super type.

File: template/java8/model/src/main/java/com/openfaas/model/Request.java (hunk 3 above)
Comment (richierich1610) [reply template/java8/model/src/main/java/com/openfaas/model/Request.java]: Parsing can be done with some utility method but URLDecoder is needed to safely decode query parameters and this won't be supported in utility method.

File: template/java8/model/src/main/java/com/openfaas/model/IRequest.java (hunk 2 above)
Comment (richierich1610) [reply template/java8/model/src/main/java/com/openfaas/model/IRequest.java]: you mean getQueryParameters() should be changed to getQuery()?

File: template/java8/model/src/main/java/com/openfaas/model/Request.java (hunk 5 above)
Comment (alexellis) [reply template/java8/model/src/main/java/com/openfaas/model/Request.java]: Maybe we could add it?

**Pull request merged**